### ISSUE TITLE
Experiencing crash when rendering Skeleton Component on high DPI monitors

### ISSUE THREAD

**Issue by UltraTempest** [Workday/canvas-kit#963]
## 🐛 Bug Report

"Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops."
Error & crash is seen when when rendering the Skeleton Component on high DPI monitors.

It seems that the componentDidUpdate function used in the Skeleton is the root of the cause: https://github.com/Workday/canvas-kit/blob/master/modules/skeleton/react/lib/skeleton.tsx

It might be caused by rounding discrepancies with HiDPI monitors.

## To Reproduce
Render the Skeleton Component on a high DPI monitor. 
The crash should be seen immediately.

## Expected Behavior

The page should not crash & no error displayed.

## Actual Results

The page crashes & an error is displayed.

## Browser (if applicable)
Chrome, Firefox & Edge

### Error Output

https://reactjs.org/docs/error-decoder.html/?invariant=185
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

**Comment by willklein** [comment]
Both the feature team and I could not reproduce the issue, but we could recognize complexities that could be alleviated. Will be addressed by #1001, which will reduce the internal state footprint and reduce potential state updates.
